File: src/main/java/com/kairlec/local/utils/PasswordCoderUtils.java
package com.kairlec.local.utils;

import com.kairlec.utils.LocalConfig;
import com.kairlec.utils.RSACoder;

import java.util.Base64;

public abstract class PasswordCoderUtils {
    private PasswordCoderUtils() {
        throw new AssertionError("工具类不允许实例化");
    }

    public static String fromRequest(String password) {
        try {
            return new String(RSACoder.decryptByPrivateKey(Base64.getDecoder().decode(password), LocalConfig.getPrivateKey()));
        } catch (Exception e) {
            e.printStackTrace();
            return password;
        }
    }


}

File: src/main/java/com/kairlec/local/utils/PasswordCoderUtils.kt
package com.kairlec.local.utils

import com.kairlec.dao.ConfigDao
import com.kairlec.utils.RSACoder
import java.util.*

object PasswordCoderUtils {
    fun fromRequest(password: String): String {
        return try {
            String(RSACoder.decryptByPrivateKey(Base64.getDecoder().decode(password), ConfigDao.Instance.privateKey))
        } catch (e: Exception) {
            e.printStackTrace()
            password
        }
    }
}
